server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -2065,10 +2071,14 @@ public boolean restartNetwork(Long networkId, boolean cleanup, boolean makeRedun
     public boolean restartNetwork(RestartNetworkCmd cmd) throws ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException {
         // This method restarts all network elements belonging to the network and re-applies all the rules
         Long networkId = cmd.getNetworkId();
+        NetworkVO network = getNetworkVO(cmd.getNetworkId(),"Nework to restart not found %s", String.valueOf(networkId));
         boolean cleanup = cmd.getCleanup();
+        if (network.getVpcId() != null && cleanup) {
```

Comment:
should remove guest nic and re-add guest nic, or forbid the behavior ?
each option has pros and cons.

if restarting vpc tier is forbidden, UI needs to be updated as well.

Reply:
Good point, I had choosen for forbidding, but this only does it at API level.
I'll look at the UI as well